src/main/java/com/fasterxml/jackson/databind/ser/BeanSerializerFactory.java

```diff
@@ -457,7 +453,31 @@ protected JsonSerializer<Object> constructBeanOrAddOnSerializer(SerializerProvid
             PropertyName name = PropertyName.construct(anyGetter.getName());
             BeanProperty.Std anyProp = new BeanProperty.Std(name, valueType, null,
                     anyGetter, PropertyMetadata.STD_OPTIONAL);
-            builder.setAnyGetter(new AnyGetterWriter(anyProp, anyGetter, anySer));
+
```

Comment:
Since we are now handling props to either replace or add any-getter, I was thinking maybe instead of.... 

```java
if (anyGetter != null) {
         // 40+ lines of code
         // many more
}
```

... this, we can do... ..

```java
     if (anyGetter != null) {
          props = updateAnyGetterProp(props);
     }
```
... like this way, simliar to `collectAll()` from `POJOPropertiesCollector` class (Or maybe I am thinking too far at this point 🤔